### ISSUE TITLE
DDLDatabase cannot handle tables containing COLLATE clauses

### ISSUE THREAD

**Issue by amir20001** [jOOQ/jOOQ#8011]
### Steps to reproduce the problem (include example code if possible):
```
CREATE TABLE `account` (
  `id` char(36) NOT NULL ,
  `name` varchar(150) COLLATE utf8_unicode_ci NOT NULL,
  PRIMARY KEY (`id`)
)
```
will generate the following error 

```
Error running jOOQ code generation tool: Error while exporting schema: SQL [create table "account"("id" char(36) not null, "name" varchar(150) collate utf8_unicode_ci not null, primary key ("id"))]; Syntax error in SQL statement "CREATE TABLE ""account""(""id"" CHAR(36) NOT NULL, ""name"" VARCHAR(150) COLLATE[*] UTF8_UNICODE_CI NOT NULL, PRIMARY KEY (""id"")) "; expected "FOR, UNSIGNED, INVISIBLE, VISIBLE, NOT, NULL, AS, DEFAULT, GENERATED, ON, NOT, NULL, AUTO_INCREMENT, BIGSERIAL, SERIAL, IDENTITY, NULL_TO_DEFAULT, SEQUENCE, SELECTIVITY, COMMENT, CONSTRAINT, PRIMARY, UNIQUE, NOT, NULL, CHECK, REFERENCES, ,, )"; SQL statement:
[ERROR] create table "account"("id" char(36) not null, "name" varchar(150) collate utf8_unicode_ci not null, primary key ("id")) [42001-197]

```

but this will generate fine
```
 CREATE TABLE `account` (
  `id` char(36) NOT NULL ,
  `name` varchar(150)  NOT NULL,
  PRIMARY KEY (`id`)
)
```
### Versions:

- jOOQ: 3.11.5
- Java: 1.8.0_181-b13
- Database (include vendor): mysql
- OS: mac os 10.13.6
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks for your report. I'm assuming you're using the `DDLDatabase` on this statement?

**Comment by lukaseder** [comment]
In case this is really about the `DDLDatabase`, I can reproduce it with the same error message. Will fix right away.

Due to a recent regression in 3.11.6, I will release 3.11.7 containing this fix as well on Monday next week.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.7 (#8014)
